ouroboros-consensus-shelley/src/Ouroboros/Consensus/Shelley/Protocol.hs

```diff
@@ -293,26 +296,61 @@ data instance ConsensusConfig (TPraos c) = TPraosConfig {
 -- Use generic instance
 instance TPraosCrypto c => NoUnexpectedThunks (ConsensusConfig (TPraos c))
 
+-- | Separate type instead of 'Bool' for the custom 'Ord' instance +
+-- documentation.
+data SelfIssued =
+    -- | A block we produced ourself
+    SelfIssued
+    -- | A block produced by another node
+  | NotSelfIssued
+  deriving (Show, Eq)
+
+instance Ord SelfIssued where
+  compare SelfIssued    SelfIssued    = EQ
+  compare NotSelfIssued NotSelfIssued = EQ
+  compare SelfIssued    NotSelfIssued = GT
+  compare NotSelfIssued SelfIssued    = LT
+
 -- | View of the ledger tip for chain selection.
 --
---   We order between chains as follows:
---   - By chain length, with longer chains always preferred; _else_
---   - By the leader value of the chain tip, with lower values preferred; _else_
---   - If the tip of each chain was issued by the same agent, then we prefer the
---     chain whose tip has the highest ocert issue number, if one exists; _else_
---   - All chains are considered equally preferable
-data TPraosChainSelectView c = ChainSelectView {
+-- We order between chains as follows:
+--
+-- 1. By chain length, with longer chains always preferred.
+-- 2. If the tip of each chain has the same slot number, we prefer the one tip
+--    that we produced ourselves.
+-- 3. If the tip of each chain was issued by the same agent, then we prefer
+--    the chain whose tip has the highest ocert issue number.
+-- 4. By the leader value of the chain tip, with lower values preferred.
```

Comment:
If the tips are produced by the same agent, then I would expect them to have identical leader values in the same slot. So I guess this change is to prefer chains with tips in different slots?

Reply:
Ah, because the VRF doesn't change when a new ocert is issued.

So the previous order was correct, but the new one is also correct, as it doesn't matter in the end. I find this order a bit easier to understand, though: as the last tie breaker, use the leader value, all other comparisons are more important.